feat: create TableHeadRow component

raky291 opened the pull request:
- [x] I have followed the [Your First Code Contribution section of the Contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md#your-first-code-contribution)

Summarize the changes made and the motivation behind them.

Reference related issues using `#` followed by the issue number.

If there are breaking API changes - like adding or removing props, or changing the structure of the theme - describe them, and provide steps to update existing code.

---

https://github.com/themesberg/flowbite-react/issues/1467

Please review my code, I'm not sure if I need to modify more code. Regards.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
  - Introduced `HeadRow` to the Table component for improved header row control.
  - Added a new `TableHeadRow` component for enhanced customization of table header rows.

- **Improvements**
  - Enhanced theming capabilities of table headers with a new `row` property for better style customization.
  - Improved structural organization of table headers through the use of `TableHeadRow` in tests.
  - Updated contribution guidelines to include a new step for creating changesets before pull requests.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rluders commented:
@raky291 this looks good, but I think that we will introduce some breaking changes. Can you just adjust the PR and maybe the commit to inform it? - this kind of information is added to the release notes/changelog. :)

raky291 commented:
@rluders Thank you for checking my PR, I have already uploaded my changes and added the changeset, I also added some changes to CONTRIBUTING.md I hope that is okay. Greetings.

rluders commented:
@raky291 could you please just revert this https://github.com/themesberg/flowbite-react/pull/1470/commits/0ffc988f3c5df93f44de363e026f10eaa6ca6b8b and instead of using merge use rebase?

After reverting, inside your branch:

```sh
git pull origin main --rebase
```

SutuSebastian commented:
This needs a re-sync with the `main` branch + fix conflicts
